src/foam/u2/view/RichTextArea.js

```diff
@@ -0,0 +1,329 @@
+          n.el().contentDocument.body.style.overflow = 'auto';
+          n.el().contentDocument.body.style.margin = '5px';
+          n.el().contentDocument.body.contentEditable = true;
+          n.el().contentDocument.body.insertAdjacentHTML('beforeend', self.data);
+          n.el().contentDocument.body.addEventListener('input', function () {
+            self.data = n.el().contentDocument.body.innerHTML;
+          }, false);
+
+        });
+      }
+    },
+  ],
+
+  methods: [
+    function initE() {
+      this.SUPER();
+      this
+        .start('div').addClass('foam-flow-Document')
+          .start('iframe', null, this.frame$).addClass('iframeContainer')
+            .attrs({
+              id: 'frame',
+              name: 'frame',
+              frameBorder: 0,
+              sandbox: 'allow-same-origin'
+            })
+          .end()
+          .startContext({ data: this })
+            .start('div', null, this.richActions$).addClass('richtext-actions')
+              .start(this.BOLD).addClass('richtext-button').end()
+              .start(this.ITALIC).addClass('richtext-button').end()
+              .start(this.UNDERLINE).addClass('richtext-button').end()
+              .start('div').addClass('dropdown')
+                .start(this.FONT_SIZE).addClass('richtext-button').end()
+                .start(this.FontSize.create({}), null, this.fontSizeMenu$).end()
+              .end()
+              .start('div').addClass('dropdown')
+                .start(this.FONT_FACE).addClass('richtext-button').end()
+                .start(this.FontFace.create({}), null, this.fontFaceMenu$).end()
+              .end()
+              .start('div').addClass('dropdown')
+                .start(this.JUSTIFICATION).addClass('richtext-button').end()
+                .start(this.TextAlignment.create({}), null, this.justifyMenu$).end()
+              .end()
+              .start('div').addClass('dropdown')
+                .start(this.OTHERS).addClass('richtext-button').end()
+                .start(this.TextFormat.create({}), null, this.otherActionsMenu$).end()
+              .end()
+            .end()
+          .endContext()
+
+          .start(this.DetailView, {data: this.FontFace2.create(), showActions: true}).end()
+          .tag(this.Popup, {
+            button: 'CLICK',
+            view: this.DetailView.create({data: this.FontSize2.create(), showActions: true})
+          })
+        .end();
+    },
+
+    function closeOpenMenu(n) {
+      var actions = this.richActions.children;
+      for (i = 0; i < actions.length; i++) {
+        if ( actions[i].nodeName === 'DIV' ) {
+          if ( actions[i].children[1].menu.id === n.id ) {
+            if ( n.shown == true ) {
+              n.hide();
+            } else { n.show(); }
+          } else {
+            actions[i].children[1].menu.hide();
+          }
+        }
+      }
+    }
+  ],
+
+  actions: [
+    {
```

Comment:
Do you need these actions still? I think you pulled all of these out into their own components.

Reply:
No, would remove it.